### PR TITLE
chore: Devcontainer upgrade to Ubuntu Noble & Node 22

### PR DESCRIPTION
**Pull request by GingerGeek**
## Pre-flight checklist

- [X] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [X] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [X] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Upgrades Dev Container to Ubuntu 24.04 base and Node 22 (latest LTS)

Also migrates VSCode configuration/settings to correct place in schema

Adds a fairly hacky workaround for #10769 where if the `start` or `serve` commands are ran within Remote Containers it will default to `127.0.0.1` instead of `localhost`. Ideally this would have been a cleaner refactor, or fixed somewhere upstream (ie vscode forwards port correctly, or webpack dev server support dual-stack binding)

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs

#10769

### CODE REVIEW

**Comment by slorber** [comment]
We don't want to add Devcontainer specific things into Docusaurus core.

At best I can propose to implement `docusaurus start --host 127.0.0`, it's already supported for `docusaurus serve`. Devcontainer hacks should be added "from the outside"

File: packages/docusaurus/src/server/remoteContainersHook.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      );
+    } else {
+      logger.warn('VSCode Remote may not support IPv6');
+    }
+  }
+}
+
+export type OptionallyProvidedHost = {
+  host?: string;
+};
```
Comment (slorber) [inline packages/docusaurus/src/server/remoteContainersHook.ts:27]: We don't want to add Devcontainer specific things into Docusaurus core.

At best I can propose to implement `docusaurus start --host 127.0.0`, it's already supported for `docusaurus serve`. Devcontainer hacks should be added "from the outside"

**Review by slorber — CHANGES_REQUESTED** [review]

File: packages/docusaurus/src/server/remoteContainersHook.ts (hunk 1 above)
Comment (GingerGeek) [reply packages/docusaurus/src/server/remoteContainersHook.ts:27]: I will pull out the hack from this PR so it's just the devcontainer upgrade

**Review by slorber — APPROVED** [review]
thanks 👍

**Pull request merged**